### PR TITLE
RIA-7781 - New Application type  Change hearing type

### PR DESCRIPTION
**Pull request by robin-ieong**
(https://tools.hmcts.net/jira/browse/RIA-7781)

### Checklist

- [ x] commit messages are meaningful and follow good commit message guidelines
- [x ] README and other documentation has been updated / added (if needed)
- [x ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/entities/MakeAnApplicationTypes.java [hunk 1]
```diff
@@ -17,7 +17,8 @@ public enum MakeAnApplicationTypes {
     UPDATE_HEARING_REQUIREMENTS("Update hearing requirements"),
     UPDATE_APPEAL_DETAILS("Update appeal details"),
     REINSTATE("Reinstate an ended appeal"),
-    OTHER("Other");
+    OTHER("Other"),
```
Comment (rajeshthuraiyur) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/entities/MakeAnApplicationTypes.java:20]: I would prefer to move this one above the OTHER type, as you know OTHER would come always at the end when they don't see anything that suits what they want.

**Review by Marcellofabbri — COMMENTED** [review]
Unrelated to this PR, but it's worth checking if a PR needs to be opened for ia-case-notifications-api to add the new constant to the `MakeAnApplicationTypes` enum in that repository too. Some events that involve sending notifications may break if the application type is `changeHearingType` and the notification service doesn't include that value

**Comment by robin-ieong** [comment]
> Unrelated to this PR, but it's worth checking if a PR needs to be opened for ia-case-notifications-api to add the new constant to the `MakeAnApplicationTypes` enum in that repository too. Some events that involve sending notifications may break if the application type is `changeHearingType` and the notification service doesn't include that value

Good point. I've debugged notification-api and can confirm it's using the BaseNotificationPersonalisation which doesn't require mapping the application type. There is no requirement to change the notification in this story but I've given a heads up to Simon to see if a customised notification is required like some of the other application types (if there is I expect it to be in a separate story). Thanks.

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/entities/MakeAnApplicationTypes.java (hunk 1 above)
Comment (robin-ieong) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/entities/MakeAnApplicationTypes.java]: updated, thanks

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/service/MakeAnApplicationTypesProvider.java [hunk 2]
```diff
@@ -126,6 +128,7 @@ public DynamicList getMakeAnApplicationTypes(Callback<AsylumCase> callback) {
                 values.add(new Value(LINK_OR_UNLINK.name(), LINK_OR_UNLINK.toString()));
                 values.add(new Value(JUDGE_REVIEW.name(), JUDGE_REVIEW.toString()));
                 values.add(new Value(OTHER.name(), OTHER.toString()));
+                values.add(new Value(CHANGE_HEARING_TYPE.name(), CHANGE_HEARING_TYPE.toString()));
```
Comment (Marcellofabbri) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/service/MakeAnApplicationTypesProvider.java:131]: I just realized that `CHANGE_HEARING_TYPE` is missing from the `LISTING` block. It escaped me earlier when I checked the PR

**Review by Marcellofabbri — CHANGES_REQUESTED** [review]

**Review by Marcellofabbri — APPROVED** [review]

**Review by rajeshthuraiyur — APPROVED** [review]
LGTM

**Pull request merged**